Source/WebCore/platform/network/BlobResourceHandle.cpp

```diff
@@ -220,10 +220,40 @@ void BlobResourceHandle::doStart()
         Ref<BlobResourceHandle> protectedThis(*this); // getSizeForNext calls the client
         for (size_t i = 0; i < m_blobData->items().size() && !erroredOrAborted(); ++i)
             getSizeForNext();
+        if (auto error = seek()) {
+            notifyFail(*error);
+            return;
+        }
         notifyResponse();
     }
 }
 
+auto BlobResourceHandle::adjustAndValidateRangeBounds() -> std::optional<Error>
+{
+    if (!m_rangeStart) {
+        if (!m_rangeEnd)
+            return Error::RangeError;
+        // m_rangeEnd indicates the last bytes to read.
+        if (*m_rangeEnd > m_totalSize) {
+            m_rangeStart = 0;
+            m_rangeEnd = m_totalSize ? (m_totalSize - 1) : 0;
```

Comment:
Why the -1?

I know it's not new, just moved code. But mystified; is the end the offset _before_ the last byte?

Reply:
Those are indexes. For example, if the clients want to fetch the first 1024 bytes, it should use this range header value: `bytes=0-1023`. It's zero-based. In this example, `m_rangeStart` is `0` and `m_rangeEnd` is `1023`. So I believe it is correct to set `m_rangeEnd` to `m_totalSize-1` when we return the rest of the data (till the end of the blob).